src/topology/gluing.lean

```diff
@@ -0,0 +1,255 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.category.Top
+import category_theory.glue_data
+
+/-!
+# Gluing Topological spaces
+
+Given a family of gluing data (see `category_theory/glue_data`), we can then glue them together.
+
+The construction should be "sealed" and considered as a black box, while only using the API
+provided.
+
+## Main definitions
+
+* `Top.glue_data`: A structure containing the family of gluing data.
+* `category_theory.glue_data.glued`: The glued topological space.
+    This is defined as the multicoequalizer of `∐ V i j ⇉ ∐ U i`, so that the general colimit API
+    can be used.
+* `category_theory.glue_data.ι`: The immersion `ι i : U i ⟶ glued` for each `i : ι`.
+* `Top.glue_data.rel`: A relation on `Σ i, D.U i` defined by `⟨i, x⟩ ~ ⟨j, y⟩` iff
+    `⟨i, x⟩ = ⟨j, y⟩` or `t i j x = y`. See `Top.glue_data.ι_eq_iff_rel`.
+
+## Main results
+
+* `Top.glue_data.is_open_iff`: A set in `glued` is open iff its preimage along each `ι i` is
+    open.
+* `Top.glue_data.ι_jointly_surjective`: The `ι i`s are jointly surjective.
+* `Top.glue_data.rel_equiv`: `rel` is an equivalence relation.
+* `Top.glue_data.ι_eq_iff_rel`: `ι i x = ι j y ↔ ⟨i, x⟩ ~ ⟨j, y⟩`.
+* `Top.glue_data.image_inter`: The intersection of the images of `U i` and `U j` in `glued` is
+    `V i j`.
+* `Top.glue_data.preimage_range`: The preimage of the image of `U i` in `U j` is `V i j`.
+* `Top.glue_data.preimage_image_eq_preimage_f`: The preimage of the image of some `U ⊆ U i` is
+    given by the preimage along `f j i`.
+* `Top.glue_data.ι_open_embedding`: Each of the `ι i`s are open embeddings.
+
```

Comment:
```suggestion
## Related results

There is also `topology/metric_space/gluing.lean` where we glue two metric spaces with an isometric
subspace together as a quotient type of the disjoint union.
```

Not sure what else I could say about the similarities / differences between these two constructions since they seem to be in quite tangential directions.